ui/src/app/cohort-search/cohort-search/cohort-search.component.ts

```diff
@@ -48,25 +47,23 @@ export class CohortSearchComponent implements OnInit, OnDestroy {
     console.log(`Entering CohortSearchComponent.ngOnInit with route:`);
     console.dir(this.route);
 
-    const {queryParams: query$, data: data$} = this.route;
-    this.subscription = Observable.combineLatest(query$, data$).subscribe(([params, data]) => {
-      /* EVERY time the route changes, reset the store first */
-      this.actions.resetStore();
-      this.actions.cdrVersionId = data.workspace.cdrVersionId;
+    const params = flattenedRouteQueryParams(this.route);
```

Comment:
I don't think this is equivalent functionality.  This behavior will reset the redux state with every time the component is instantiated.  The desired behavior is to reset the redux store on _every change_ to the route's data and query param objects, because otherwise [data could leak from cohort to cohort](https://angular.io/guide/router#observable-parammap-and-component-reuse).  We actually had that problem at one point, and this subscription is what guards against it.  

EDIT: I may have inadvertently made all this unnecessary though by making the `cohort-search` module lazy-loaded: the component re-use rules are different for lazy loaded vs eagerly loaded modules, and it looks like, in practice, `CohortSearchComponent` is now re-instantiated each time its routed to anyways.  

Either way, I think the only change actually needed here is this:

```
const {queryParams: query$, data: data$} = this.route.parent;
```

Reply:
Yeah its not great b/c its pretty fragile with respect to where things happen in the routing table. Also how did it skip the parent?  You only added one component but now we have to go back two to get to the data... seems weird.

Anyway the important thing is to make sure that this chunk of code is reliably run on route change, and I'd rather it be explicitly tied to the route than rely on the router never reusing this component, which seems less obviously certain.  So lets use the observables here; whether we get them from the grandparent or by walking the pathFromRoot is less of a concern to me.